docker: switch to using ubi9/ubi-minimal

jlinder opened the pull request:
Before this change, cockroach docker images were based on the ubi8/ubi-minimal image.

Now, cockroach images are based on the ubi9/ubi-minimal image.

Fixes: #112670

Release note (general change): The cockroach docker image is now based on Red Hat's ubi9/ubi-minimal image (instead of ubi8/ubi-minimal).

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/112930)
<!-- Reviewable:end -->

jlinder committed:
docker: switch to using ubi9/ubi-minimal

Before this change, cockroach docker images were based on the
ubi8/ubi-minimal image.

Now, cockroach images are based on the ubi9/ubi-minimal image.

Fixes: #112670

Release note (general change): The cockroach docker image is now based
on Red Hat's ubi9/ubi-minimal image (instead of ubi8/ubi-minimal).

rail reviewed (approved):
> Reviewed 4 of 4 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/112930)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @jlinder)



<!-- Sent from Reviewable.io -->

jlinder commented:
TFTR!

bors r=rail
